tdrs-backend/tdpservice/parsers/test/test_case_consistency.py

```diff
@@ -0,0 +1,638 @@
+"""Test the CaseConsistencyValidator."""
+
+import pytest
+import logging
+from tdpservice.parsers.test import factories
+from .. import schema_defs, util
+from ..case_consistency_validator import CaseConsistencyValidator
+from tdpservice.parsers.models import ParserErrorCategoryChoices
+
+
+logger = logging.getLogger(__name__)
+
+
+class TestCaseConsistencyValidator:
+    """Test case consistency (cat4) validators."""
+
+    def parse_header(self, datafile):
+        """Parse datafile header into header object."""
+        rawfile = datafile.file
+
+        # parse header, trailer
+        rawfile.seek(0)
+        header_line = rawfile.readline().decode().strip()
+        return schema_defs.header.parse_and_validate(
+            header_line,
+            util.make_generate_file_precheck_parser_error(datafile, 1)
+        )
+
+    @pytest.fixture
+    def tanf_s1_records(self):
+        """Return group of TANF Section 1 records."""
+        t1 = factories.TanfT1Factory.create()
+        t2 = factories.TanfT2Factory.create()
+        t3 = factories.TanfT3Factory.create()
+        t3_1 = factories.TanfT3Factory.create()
+        return [t1, t2, t3, t3_1]
+
+    @pytest.fixture
+    def tanf_s1_schemas(self):
+        """Return group of TANF Section 1 schemas."""
+        s1 = schema_defs.tanf.t1.schemas[0]
+        s2 = schema_defs.tanf.t2.schemas[0]
+        s3 = schema_defs.tanf.t3.schemas[0]
+        return [s1, s2, s3, s3]
+
+    @pytest.fixture
+    def small_correct_file(self, stt_user, stt):
```

Comment:
we are using this fixture in test_parse, maybe it's time to create "factory as fixture"

Reply:
this is from other tests that i moved into this file for consistency. i generally agree with your point, though, and think we should add something to #2641 to capture it